src/widget/number-input/decimal-input.js

```diff
@@ -27,28 +27,28 @@ const getDecimalCharacters = (languages) => {
     return characters;
 };
 
-/** @type {Map<string, RegExp>} */
-let characterPatternsByLanguage = new Map();
+/** @type {Map<string, Set<string>>} */
+let validCharactersByLanguage = new Map();
 
 /**
  * @param {string[]} languages
  */
-const getCharacterPattern = (languages) => {
+const getValidCharacters = (languages) => {
     const [language] = languages;
-    let characterPattern = characterPatternsByLanguage.get(language);
+    let validCharacters = validCharactersByLanguage.get(language);
 
-    if (characterPattern != null) {
-        return characterPattern;
+    if (validCharacters != null) {
+        return validCharacters;
     }
 
-    const decimalCharacters = getDecimalCharacters(language);
+    const locales = Intl.getCanonicalLocales(languages);
+    const formatter = Intl.NumberFormat(locales);
+    const number = -9.012345678;
 
-    characterPattern = new RegExp(
-        `[-0-9${Array.from(decimalCharacters).join('')}]`
-    );
-    characterPatternsByLanguage.set(language, characterPattern);
+    validCharacters = new Set(`${number}${formatter.format(number)}`.split(''));
```

Comment:
Initially I thought there might be a change in behavior because this was not using `decimalCharacters`. But now I see both `decimalCharacters` and this include the `.` (here because `${number}` is included) and only one other decimal marker. Maybe there's an opportunity to reconcile those two ways of determining the set of allowed decimal markers? Doesn't feel like a big deal but it was the one thing that made me have to think through this change.